### PR TITLE
dnn: add ONNX TopK

### LINKED ISSUE #22890: OpenCV cannot import ONNX model: Can't create layer "..." of type "TopK"
### System Information

OpenCV => Python opencv-python-rolling 4.6.0.20221112 (no update available)
Operating System / Platform Windows 10 64 bit
Python =>3.10.4

### Detailed description

I converted the PointRend from PaddleSeg to ONNX:

https://github.com/PaddlePaddle/PaddleSeg/tree/release/2.7/configs/pointrend

This fails to load in OpenCV 4.6 pre-release:
````
[ERROR:0@0.392] global D:\a\opencv-python\opencv-python\opencv\modules\dnn\src\onnx\onnx_importer.cpp (1053) cv::dnn::dnn4_v20220524::ONNXImporter::handleNode DNN/ONNX: ERROR during processing node with 2 inputs and 2 outputs: [TopK]:(onnx_node!p2o.TopK.0) from domain='ai.onnx'
Traceback (most recent call last):
  File "d:\Local\devel\Python\OpenCV\dnn_segmentation_paddle_pointrend_voc\inference.py", line 131, in <module>
    model = cv2.dnn.readNet(model_path)
cv2.error: OpenCV(4.6.0-dev) D:\a\opencv-python\opencv-python\opencv\modules\dnn\src\onnx\onnx_importer.cpp:1072: error: (-2:Unspecified error) in function 'cv::dnn::dnn4_v20220524::ONNXImporter::handleNode'
> Node [TopK@ai.onnx]:(onnx_node!p2o.TopK.0) parse error: OpenCV(4.6.0-dev) d:\a\opencv-python\opencv-python\opencv\modules\dnn\src\net_impl.hpp:107: error: (-2:Unspecified error) Can't create layer "onnx_node!p2o.TopK.0" of type "TopK" in function 'cv::dnn::dnn4_v20220524::Net::Impl::getLayerInstance'
````
Same result with the model after onnx simplify.

### Steps to reproduce

```
model = cv2.dnn.readNet(model_path)
```
Find the ONNX file here: https://drive.google.com/file/d/1rKcUtoncMvorMtk6LdC6_QK2PSRS_pee/view?usp=sharing

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [X] I updated to the latest OpenCV version and the issue is still there
- [X] There is reproducer code and related data files (videos, images, onnx, etc)

### PR DESCRIPTION
**Pull request by fengyuentau**
Merge with https://github.com/opencv/opencv_extra/pull/1200

Partially fixes #22890 and #20258

To-do:

- [x] TopK forward impl
- [x] add tests
- [x] support Opset 1 & 10 if possible
- [ ] ~Support other backends~ (TopK has two outputs, which is not supported by other backends, such as openvino)

Perf:

M1 (time in millisecond)

| input shape     | axis | dnn  | ort  |
| --------------- | ---- | ---- | ---- |
| (1000, 100)     | 0    | 1.68 | 4.07 |
| (1000, 100) K5  | 0    | 1.13 | 0.12 |
| (1000, 100)     | 1    | 0.96 | 0.77 |
| (100, 100, 100) | 0    | 10.00 | 31.13 |
| (100, 100, 100) | 1    | 7.33 | 9.17 |
| (100, 100, 100) | 2    | 7.52 | 9.48 |

M2 (time in milisecond)

| input shape     | axis | dnn  | ort  |
| --------------- | ---- | ---- | ---- |
| (1000, 100)     | 0    | 0.76 | 2.44 |
| (1000, 100) K5 | 0 | 0.68 | 0.07 |
| (1000, 100)     | 1    | 0.41 | 0.50 |
| (100, 100, 100) | 0    | 4.83 | 17.52|
| (100, 100, 100) | 1    | 3.60 | 5.08 |
| (100, 100, 100) | 2    | 3.73 | 5.10 |

ONNXRuntime performance testing script: https://gist.github.com/fengyuentau/a119f94fd16721ec9974b8c7b0a45d4c

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by Abdurrahheem** [comment]
Hi! 

Could you please explain why this PR has been temporarily halted ?

**Comment by fengyuentau** [comment]
CPU implementation is done.

@asmorkalov I will prepare a dedicated PR to merge it to 5.x since one of the ouputs is of type CV_32S which cannot handled in 4.x.

File: modules/dnn/perf/perf_layer.cpp [hunk 1]
```diff
@@ -1043,4 +1043,64 @@ INSTANTIATE_TEST_CASE_P(/**/, Layer_Elementwise,
...
+    test_layer(input_shape_2d, input_shape_2d[1] / 2, 1);
+}
+PERF_TEST_P_(Layer_TopK, TopK_3D_Axis0) {
+    test_layer(input_shape_3d, input_shape_3d[0] / 2, 0);
+}
+PERF_TEST_P_(Layer_TopK, TopK_3D_Axis1) {
+    test_layer(input_shape_3d, input_shape_3d[1] / 2, 1);
+}
+PERF_TEST_P_(Layer_TopK, TopK_3D_Axis2) {
+    test_layer(input_shape_3d, input_shape_3d[2] / 2, 2);
```
Comment (asmorkalov) [inline modules/dnn/perf/perf_layer.cpp:1094]: I propose to add cases for top-5 at least. It's popular case for classification scenarios.

Comment (fengyuentau) [reply modules/dnn/perf/perf_layer.cpp:1097]: Added

**Comment by asmorkalov** [comment]
@dkurt could you take a look too?

**Comment by mshabunin** [comment]
Looks like `TopK` test is failing on CI now. Strangely it did not fail in this PR :leftwards_hand: :confused: :rightwards_hand:

**Comment by fengyuentau** [comment]
> Looks like `TopK` test is failing on CI now. Strangely it did not fail in this PR 🫲 😕 🫱

@mshabunin Merge of this PR and its dependent in opencv_extra had a two-hour gap. Most likely this is the reason why it fails. Example: https://pullrequest.opencv.org/buildbot/builders/precommit_linux64/builds/108617.

**Comment by asmorkalov** [comment]
It's my fault, sorry guys. I missed extra part of the patch.

**Pull request merged**